xdsl/dialects/riscv.py

```diff
@@ -887,6 +1085,24 @@ def verify_(self) -> None:
     def assembly_line_args(self) -> tuple[_AssemblyInstructionArg, ...]:
         return self.rd, self.csr, self.rs1
 
+    def print_attributes(self, printer: Printer, first: bool) -> None:
+        printer.print(" " if first else ", ")
+        printer.print(self.csr.value.data)
+        if self.writeonly is not None:
+            printer.print(", w")
+
+    @classmethod
+    def parse_attributes(cls, parser: Parser) -> dict[str, Attribute]:
+        attributes = dict[str, Attribute]()
+        if (csr := parser.parse_optional_integer()) is not None:
+            attributes["csr"] = IntegerAttr(csr, i32)
+        else:
+            raise NotImplementedError()
```

Comment:
Here I would instead suggest using `parser.parse_integer(allow_boolean=False, allow_negative=False)` here instead of optional.

Reply:
Done